resources/bundles/org.eclipse.core.resources/src/org/eclipse/core/internal/events/AutoBuildJob.java

```diff
@@ -95,6 +95,7 @@ synchronized void build(boolean needsBuild) {
 		//don't mess with the interrupt flag if the job is still running
 		if (state != Job.RUNNING)
 			setInterrupted(false);
```

Comment:
It feels wrong, that this simple check is unrelated too all those more complicated conditions below.  is it because not running implies current thread is not the autobuild?

Reply:
See bug 164496